src/pydiverse/pipedag/backend/table/sql/sql.py

```diff
@@ -1278,48 +1280,49 @@ class TableReference:
     """Reference to a user-created table.
 
     By returning a `TableReference` wrapped in a :py:class:`~.Table` from a task,
-    you can tell pipedag that you yourself created a new table inside
-    the correct schema of the table store.
-    This may be useful if you need to perform a complex load operation to create
-    a table (e.g. load a table from an Oracle database into Postgres
-    using `oracle_fdw`).
+    you can tell pipedag about a table or a view in an `external_schema`.
 
     Only supported by :py:class:`~.SQLTableStore`.
 
     Warning
     -------
-    Using table references is not recommended unless you know what you are doing.
-    It may lead unexpected behaviour.
-    It also requires accessing non-public parts of the pipedag API which can
-    change without notice.
+    When using a `TableReference`, pipedag has no way of knowing the cache validity
+    of the external object. Hence, the user should provide a cache function for the
+    `Task` or version the `Task`.
+    It is now allowed to specify a `TableReference` to a table in schema of the
+    current stage.
 
     Example
     -------
-    Making sure that the table is created in the correct schema is not trivial,
-    because the schema names usually are different from the stage names.
-    To get the correct schema name, you must use undocumented and non-public
-    parts of the pipedag API. To help you get started, we'll provide you with
-    this example, however, be warned that this might break without notice::
+    You can use a `TableReference` to tell pipedag about a table that exists
+    in an external schema::
 
         @materialize(version="1.0")
         def task():
-            from pydiverse.pipedag.context import ConfigContext, TaskContext
+            return Table(TableReference("external_schema"), "name_of_table")
```

Comment:
We might rename TableReference since it is tied to external and we might forbid the use of the name != None in dag.Table.
```suggestion
            return Table(ExternalTableReference("name_of_table", schema="external_schema"))
```